infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -573,7 +582,7 @@ private void runFailureDetectorTask() {
     private Set<String> getNewFailures(PollReport pollReport) {
         return Sets.difference(
                 pollReport.getFailingNodes(),
-                new HashSet<>(serverContext.getManagementLayout().getUnresponsiveServers()));
+                new HashSet<>(getManagementLayout().getUnresponsiveServers()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getManagementLayout()" can return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)